ethcore/src/account.rs

```diff
@@ -120,7 +120,7 @@ impl Account {
 	}
 
 	/// Get (and cache) the contents of the trie's storage at `key`.
-	pub fn storage_at(&self, db: &AccountDB, key: &H256) -> H256 {
+	pub fn storage_at(&self, db: &HashDB, key: &H256) -> H256 {
```

Comment:
passing anything other than an AccountDB could introduce bugs or database key conflicts. imo it is better to leverage a strong type system to preserve invariants like that